include/picolibrary/microchip/megaavr0/peripheral/bod.h

```diff
@@ -0,0 +1,436 @@
+
+namespace picolibrary::Microchip::megaAVR0::Peripheral {
+
+/**
+ * \brief Microchip megaAVR 0-series Brown-Out Detector (BOD) peripheral.
+ */
+class BOD {
+  public:
+    /**
+     * \brief Control A (CTRLA) register.
+     *
+     * This register has the following fields:
+     * - Sleep (SLEEP)
+     * - Active (ACTIVE)
+     * - Sample Frequency (SAMPFREQ)
+     */
+    class CTRLA : public Protected_Register<std::uint8_t> {
+      public:
+        /**
+         * \brief Field sizes.
+         */
+        struct Size {
+            static constexpr auto SLEEP     = std::uint_fast8_t{ 2 }; ///< SLEEP.
+            static constexpr auto ACTIVE    = std::uint_fast8_t{ 2 }; ///< ACTIVE.
+            static constexpr auto SAMPFREQ  = std::uint_fast8_t{ 1 }; ///< SAMPFREQ.
+            static constexpr auto RESERVED5 = std::uint_fast8_t{ 3 }; ///< RESERVED5.
+        };
+
+        /**
+         * \brief Field bit positions.
+         */
+        struct Bit {
+            static constexpr auto SLEEP = std::uint_fast8_t{}; ///< SLEEP.
+            static constexpr auto ACTIVE = std::uint_fast8_t{ SLEEP + Size::SLEEP }; ///< ACTIVE.
+            static constexpr auto SAMPFREQ = std::uint_fast8_t{ ACTIVE + Size::ACTIVE }; ///< SAMPFREQ.
+            static constexpr auto RESERVED5 = std::uint_fast8_t{ SAMPFREQ + Size::SAMPFREQ }; ///< RESERVED5.
+        };
+
+        /**
+         * \brief Field bit masks.
+         */
+        struct Mask {
+            static constexpr auto SLEEP = mask<std::uint8_t>( Size::SLEEP, Bit::SLEEP ); ///< SLEEP.
+            static constexpr auto ACTIVE = mask<std::uint8_t>( Size::ACTIVE, Bit::ACTIVE ); ///< ACTIVE.
+            static constexpr auto SAMPFREQ = mask<std::uint8_t>( Size::SAMPFREQ, Bit::SAMPFREQ ); ///< SAMPFREQ.
+            static constexpr auto RESERVED5 = mask<std::uint8_t>( Size::RESERVED5, Bit::RESERVED5 ); ///< RESERVED5.
+        };
+
+        /**
+         * \brief SLEEP.
+         */
+        enum SLEEP : std::uint8_t {
+            SLEEP_DIS     = 0x0 << Bit::SLEEP, ///< Disabled.
+            SLEEP_ENABLED = 0x1 << Bit::SLEEP, ///< Enabled in continuous mode.
+            SLEEP_SAMPLED = 0x2 << Bit::SLEEP, ///< Enabled in sampled mode.
+        };
+
+        /**
+         * \brief ACTIVE.
+         */
+        enum ACTIVE : std::uint8_t {
+            ACTIVE_DIS     = 0x0 << Bit::ACTIVE, ///< Disabled.
+            ACTIVE_ENABLED = 0x1 << Bit::ACTIVE, ///< Enabled in continuous mode.
+            ACTIVE_SAMPLED = 0x2 << Bit::ACTIVE, ///< Enabled in sampled mode.
+            ACTIVE_ENWAKE  = 0x3 << Bit::ACTIVE, ///< Enabled in continuous mode. Execution is halted at wake-up until BOD is running.
+        };
+
+        /**
+         * \brief SAMPFREQ.
+         */
+        enum SAMPFREQ : std::uint8_t {
+            SAMPFREQ_1KHZ  = 0x0 << Bit::SAMPFREQ, ///< 1 kHz.
+            SAMPFREQ_125HZ = 0x1 << Bit::SAMPFREQ, ///< 125 Hz.
+        };
+
+        CTRLA() = delete;
```

Comment:
Enum is not defined by the datasheet.
```suggestion
        };

        CTRLA() = delete;
```